pallets/ddc-clusters/src/lib.rs

```diff
@@ -394,6 +395,55 @@ pub mod pallet {
 
 			Self::do_validate_node(cluster_id, node_pub_key, succeeded)
 		}
+
+		#[pallet::call_index(5)]
+		#[pallet::weight(<T as pallet::Config>::WeightInfo::join_cluster())]
+		pub fn join_cluster(
```

Comment:
As we introduced the `join_cluster` extrinsic, let's remove checks and calls to the `node_provider_auth_contract_address` within the `add_node` extrinsic body to eliminate ambiguities of the `node_provider_auth_contract` main purpose.

The main goal of this contract is to check whether a node provided is allowed to join a cluster without waiting for action from the cluster manager. In the scope of the `add_node` extrinsic, the contract validation doesn't make sense as the Cluster Manager is permitted to change the contract (address) whenever it wants to. So the checks in the contract should be performed only in the scope of `join_cluster` extrinsic

Reply:
The `node_provider_auth_contract` call removed from `add_node`.